modules/mysql/mysql.go

```diff
@@ -0,0 +1,575 @@
+	   [innodb_rows_deleted] = [0]
+	   [innodb_rows_updated] = [346]
+	   [key_blocks_not_flushed] = [0]
+	   [binlog_cache_disk_use] = [0]
+	   [connection_errors_accept] = [0]
+	   [connection_errors_max_connections] = [0]
+	   [connection_errors_select] = [0]
+	   [innodb_buffer_pool_read_ahead_evicted] = [0]
+	   [innodb_os_log_pending_writes] = [0]
+	   [binlog_stmt_cache_use] = [0]
+	   [com_update] = [0]
+	   [handler_delete] = [0]
+	   [handler_read_key] = [2363]
+	   [handler_read_rnd] = [360]
+	   [handler_read_rnd_next] = [10166]
+	   [innodb_data_pending_reads] = [0]
+	   [innodb_rows_inserted] = [0]
+	   [handler_commit] = [678]
+	   [innodb_log_write_requests] = [765]
+	   [opened_files] = [28]
+	   [questions] = [37]
+	   [select_range_check] = [0]
+	   [binlog_stmt_cache_disk_use] = [0]
+	   [innodb_buffer_pool_bytes_dirty] = [0]
+	   [innodb_buffer_pool_reads] = [950]
+	   [innodb_row_lock_current_waits] = [0]
+	   [select_scan] = [35]
+	   [table_locks_immediate] = [12]
+	   [threads_running] = [2]
+	*/
+}
+
+func slaveSeconds(value interface{}) int64 {
+	v, err := strconv.ParseInt(fmt.Sprintf("%s", value), 10, 64)
+	if err != nil {
+		return -1
+	}
+	return v
+}
+
+func slaveRunning(value interface{}) int64 {
+	if v, ok := value.(string); ok {
+		if v == "Yes" {
+			return 1
+		}
+
+		return -1
+	}
+
+	if v := fmt.Sprintf("%s", value); v == "Yes" {
+		return 1
+	}
+
+	return -1
+}
+
+var slaveStats = map[string]func(value interface{}) int64{
+	"Seconds_Behind_Master": slaveSeconds,
+	"Slave_SQL_Running":     slaveRunning,
+	"Slave_IO_Running":      slaveRunning,
+}
+
+// hasReplPriv returns if the current user has privileges for the "SHOW SLAVE STATUS" query.
+func (m *MySQL) hasReplPriv() bool {
+	if m.user == "" {
+		return false
+	}
+
+	rows, err := m.db.Query("select User,Super_priv,Repl_client_priv from mysql.user")
+	if err != nil {
+		return false
+	}
+
+	defer rows.Close()
+
+	for rows.Next() {
+		var (
+			user           string
+			superPriv      string // Y or N
+			replClientPriv string // Y or N
+		)
+
+		err = rows.Scan(&user, &superPriv, &replClientPriv)
+		if err != nil {
+			return false
+		}
+
+		// either  SUPER or REPLICATION CLIENT.
+		if user == m.user && (superPriv == "Y" || replClientPriv == "Y") {
+			return true
+		}
+	}
+
+	return false
+
+}
+
+// https://dev.mysql.com/doc/refman/8.0/en/show-slave-status.html
+func (m *MySQL) collectSlaveStatus(metrics map[string]int64) error {
+	if !m.replicationPrivileges {
+		return nil
+	}
+
+	rows, err := m.db.Query("SHOW SLAVE STATUS")
+	if err != nil {
+		return err
+	}
+
+	defer rows.Close()
+
+	for rows.Next() {
+		var (
+			varName string
+			value   interface{}
+		)
+
+		err = rows.Scan(&varName, &value)
+		if err != nil {
+			return err
+		}
+
+		// fmt.Printf("var name: %s | value: %s\n", varName, value)
+
+		if valFn, ok := slaveStats[varName]; ok { // hmmm...
+			value := valFn(value)
+			metrics[toKey(varName)] = value
+		}
+	}
+
+	return nil
```

Comment:
Doesn't work

i found https://github.com/gdaws/mysql-slave-status

> [ ERROR ] mysql[goland] logger.go:99 error on collecting slave status: sql: expected 47 destination arguments in Scan, not 2

Reply:
i did a quick check (replaced `collectSlaveStatus` code with code from `mysql-slave-status`)

it, works

`
map[Relay_Log_Pos:{1183 true} Replicate_Wild_Do_Table:{ true} Last_Errno:{0 true} Exec_Master_Log_Pos:{895 true} Until_Log_File:{ true} Last_IO_Errno:{0 true} Last_IO_Error:{ true} Master_SSL_Cipher:{ true} Replicate_Ignore_Server_Ids:{ true} Master_Server_Id:{1 true} Relay_Master_Log_File:{mysql-bin.000002 true} Replicate_Do_Table:{ true} Master_Log_File:{mysql-bin.000002 true} Master_SSL_CA_File:{ true} Master_SSL_Crl:{ true} Replicate_Do_Domain_Ids:{ true} Parallel_Mode:{conservative true} Master_SSL_Crlpath:{ true} Connect_Retry:{60 true} Replicate_Ignore_DB:{ true} Skip_Counter:{0 true} Master_SSL_Key:{ true} Seconds_Behind_Master:{0 true} Last_SQL_Errno:{0 true} Last_SQL_Error:{ true} Using_Gtid:{No true} Slave_IO_State:{Waiting for master to send event true} Master_User:{slave_user true} Replicate_Do_DB:{ true} Until_Log_Pos:{0 true} Gtid_IO_Pos:{ true} Replicate_Ignore_Domain_Ids:{ true} Master_SSL_Cert:{ true} Master_Host:{100.127.0.91 true} Read_Master_Log_Pos:{895 true} Replicate_Wild_Ignore_Table:{ true} Last_Error:{ true} Relay_Log_Space:{1833 true} Master_SSL_Allowed:{No true} Master_SSL_CA_Path:{ true} Master_Port:{3306 true} Relay_Log_File:{mysql-relay-bin.000003 true} Slave_IO_Running:{Yes true} Slave_SQL_Running:{Yes true} Replicate_Ignore_Table:{ true} Until_Condition:{None true} Master_SSL_Verify_Server_Cert:{No true}]
`